lib/config.go

```diff
@@ -545,8 +545,15 @@ func (c *RuntimeConfig) Validate(systemContext *types.SystemContext, onExecution
 // `nil`.
 func (c *NetworkConfig) Validate(onExecution bool) error {
 	if onExecution {
-		if err := utils.IsDirectory(c.NetworkDir); err != nil {
-			return errors.Wrapf(err, "invalid network_dir: %s", err)
+		err := utils.IsDirectory(c.NetworkDir)
+		if err != nil {
+			if os.IsNotExist(err) {
+				if err = os.MkdirAll(c.NetworkDir, 0755); err != nil {
+					return errors.Wrapf(err, "invalid network_dir: %s", err)
+				}
+			} else {
+				return errors.Wrapf(err, "invalid network_dir: %s", err)
```

Comment:
This path is not covered by an unit test, can you add one to `lib/config_test.go`?